### PR TITLE
RFC 3.1: Support others calenders(Persian, Islamic, Hindi,....)

### PR DESCRIPTION
**Pull request by zoghal**
support multi calendar(japanese|buddhist|chinese|persian|indian|islamic|hebrew|indian|coptic|ethiopic)
How's this? Can it be adding to the core?

### CODE REVIEW

File: src/I18n/Time.php [hunk 1]
```diff
@@ -588,6 +588,10 @@ protected function _formatObject($date, $format, $locale)
             );
         }
 
+        if(preg_match('/@calendar=(japanese|buddhist|chinese|persian|indian|islamic|hebrew|indian|coptic|ethiopic)/', $locale)) {
```
Comment (ADmad) [inline src/I18n/Time.php]: Space missing after `if`. `indian` repeated in regex.

**Comment by ADmad** [comment]
Test failing

**Comment by zoghal** [comment]
Each version has a different output!!! intl has bug?!

**Comment by lorenzo** [comment]
@zoghal unfortunately, yes. Each version of ICU contains changes, that makes it difficult to create unit tests. I like the changes, though.

File: src/I18n/Time.php [hunk 2]
```diff
@@ -588,6 +588,10 @@ protected function _formatObject($date, $format, $locale)
             );
         }
 
+        if (preg_match('/@calendar=(japanese|buddhist|chinese|persian|indian|islamic|hebrew|coptic|ethiopic)/', $locale)) {
+            static::$_formatters[$key]->setCalendar(\IntlDateFormatter::TRADITIONAL);
```
Comment (lorenzo) [inline src/I18n/Time.php]: Why do this here and not use the `$calendar` variable? in this function?

Comment (markstory) [inline src/I18n/Time.php]: Or set calendar higher up in the method so the formatter is initialised correctly.

**Comment by lorenzo** [comment]
@zoghal bump

**Comment by zoghal** [comment]
@lorenzo  Sorry for the delay in responding to you.
In Iran, again protocol  ssl is limited and proxy speed is very slow. (For nuclear agreements between Iran and group 5+1.

One question:
  If I had all the tests I apply these values to what to do.?
in app i set in bootstrap, but for tests i don't know
```
date_default_timezone_set('Asia/Tehran');
ini_set('intl.default_locale', 'fa-IR@calendar=persian');
\Cake\I18n\I18n::locale('fa-IR@calendar=persian');
\Cake\I18n\Time::$defaultLocale = 'fa-IR@calendar=persian';
```
‍‍‍

**Comment by lorenzo** [comment]
@zoghal There is only one failing test, I think you can remove it as the other tests are actually proving that it works. I don't think there are any other tests needed to get this merged

**Comment by zoghal** [comment]
@lorenzo The main problem is that I'm working in Windows. There are many differences between Windows and Linux in icu library.
If I do not find the problem, causing further problems in the field FormHelper and ORM.

**Comment by zoghal** [comment]
@lorenzo finished :smile: . this Request closing for cleaning commits and resend new PR

**Pull request closed without merging**